### PR TITLE
Remove unwanted properties from DataMigration/services

### PR DESCRIPTION
**Pull request by haitsongmsft**
We will not expose public key properties and NIC as these will not be used in template deployments, and it is created during service creation behind scene. If added into template, it will not be respected during deployment

### CODE REVIEW

File: schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json [hunk 1]
```diff
@@ -775,14 +775,6 @@
...
       "type": "object",
       "properties": {
-        "publicKey": {
-          "type": "string",
-          "description": "The public key of the service, used to encrypt secrets sent to the service"
-        },
-        "virtualSubnetId": {
-          "type": "string",
-          "description": "The ID of the Microsoft.Network/virtualNetworks/subnets resource to which the service should be joined"
-        }
```
Comment (TianoMS) [inline schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json]: removing these properties leaves an empty property bag. This can be a problem because an empty property bag can match any object.

File: schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json [hunk 2]
```diff
@@ -775,14 +775,6 @@
...
       },
       "required": [
         "virtualSubnetId"
```
Comment (TianoMS) [inline schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json]: >virtualSubnetId [](start = 9, length = 15)

property is still in required list.

Comment (haitsongmsft) [reply schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json]: Fixed, putting the pubkey back as placeholder

File: schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json (hunk 1 above)
Comment (haitsongmsft) [reply schemas/2017-11-15-privatepreview/Microsoft.DataMigration.json]: putting the pubkey back as placeholder

**Comment by haitsongmsft** [comment]
let me make another change, we don't want to give publickey in response

**Comment by Tiano2017** [comment]
Looks like all the changes are reverted. do you still need this PR?

**Comment by haitsongmsft** [comment]
Yes, I changed to remove public key only, and leave subnet alone

**Comment by haitsongmsft** [comment]
please still CR this, we need to remove public key.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 .vscode
 js-coverage
-node_modules
\ No newline at end of file
+node_modules
+/tools
```
Comment (TianoMS) [inline .gitignore]: we can't ignore everything in tools.

Comment (haitsongmsft) [reply .gitignore]: removed

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,3 @@
 .vscode
 js-coverage
-node_modules
\ No newline at end of file
+node_modules
```
Comment (Tiano2017) [inline .gitignore]: Looks like there are some white space changes? Can you revert the changes?

Comment (haitsongmsft) [reply .gitignore]: done
